central/scannerdefinitions/handler/handler.go

```diff
@@ -190,12 +196,19 @@ func (h *httpHandler) getUpdater(key string) *requestedUpdater {
 	updater, exists := h.updaters[key]
 	if !exists {
 		filePath := filepath.Join(h.onlineVulnDir, key+".zip")
-		var url string
+		buildURL := func(values []string) string {
+			return strings.Join(values, "/")
+		}
 
-		if key == mappingUpdaterKey {
-			url = strings.Join([]string{v4StorageDomain, mappingFile}, "/")
-		} else { // if it's uuid
-			url = strings.Join([]string{scannerUpdateDomain, key, scannerUpdateURLSuffix}, "/")
+		var url string
+		switch key {
+		case mappingUpdaterKey:
+			url = buildURL([]string{v4StorageDomain, mappingFile})
```

Comment:
NItpick: There is nothing special in `buildURL`, I think it's just better to call [`path.Join` directly](https://pkg.go.dev/path#Join).

Reply:
I can't recall if this implementation suffers from this, but we have found that `path.Join` doesn't always work nicely with URLs. There is [`url.JoinPath`](https://pkg.go.dev/net/url#JoinPath) which should be the URL equivalent to `path.Join`. I feel like `strings.Join` is ok since we know how the URLs are constructed, and it should be safe. I'm ok with a switch to `url.JoinPath`, as it is probably the preferred way to do it since it was introduced a few releases ago

Reply:
Somehow I used [url.JoinPath](https://pkg.go.dev/net/url#JoinPath) but things didn't work out. I can test it again.

Reply:
I searched a bit and path.Join is not recommended for URL but for path. Hopefully @jvdm is ok with keeping strings.Join.

Reply:
`url.JoinPath` is preferred, it was just "newly" introduced in `go1.19.`.